safe/gui/tools/multi_buffer_dialog.py

```diff
@@ -0,0 +1,229 @@
+        self.directory_button.clicked.connect(
+            self.on_directory_button_tool_clicked)
+        self.radius_form.valueChanged.connect(self.add_class_button_status)
+        self.class_form.textChanged.connect(self.add_class_button_status)
+        self.add_class_button.clicked.connect(
+            self.populate_hazard_classification)
+        self.add_class_button.clicked.connect(self.ok_button_status)
+        self.remove_class_button.clicked.connect(
+            self.remove_selected_classification)
+        self.remove_class_button.clicked.connect(self.ok_button_status)
+
+        # Set up things for context help
+        self.help_button = self.button_box.button(QtGui.QDialogButtonBox.Help)
+        # Allow toggling the help button
+        self.help_button.setCheckable(True)
+        self.help_button.toggled.connect(self.help_toggled)
+        self.main_stacked_widget.setCurrentIndex(1)
+
+        # Fix for issue 1699 - cancel button does noth  ing
+        cancel_button = self.button_box.button(QtGui.QDialogButtonBox.Cancel)
+        cancel_button.clicked.connect(self.reject)
+        # Fix ends
+        ok_button = self.button_box.button(QtGui.QDialogButtonBox.Ok)
+        ok_button.clicked.connect(self.accept)
+
+    def accept(self):
+        """Process the layer for multi buffering and generate a new layer.
+
+        .. note:: This is called on OK click.
+        """
+        # set parameter from dialog
+        input_layer = self.layer.currentLayer()
+        output_path = self.output_form.text()
+        file_name = os.path.splitext(output_path)[0].split('/')[-1]
+        radius = self.get_classification()
+        # monkey patch keywords so layer works on multi buffering function
+        input_layer.keywords = {'inasafe_fields': {}}
+
+        # run multi buffering
+        self.output_layer = multi_buffering(input_layer, radius)
+
+        # save output layer to datastore
+        data_store = Folder(os.path.dirname(output_path))
+        if self.output_form.text().endswith('.shp'):
+            data_store.add_layer(self.output_layer, file_name)
+        elif self.output_form.text().endswith('.geojson'):
+            data_store.default_vector_format = 'geojson'
+            data_store.add_layer(self.output_layer, file_name)
+
+        # add output layer to map canvas
+        self.output_layer = data_store.layer(file_name)
+        QgsMapLayerRegistry.instance().addMapLayers(
+            [self.output_layer])
+        self.done(QtGui.QDialog.Accepted)
+
+    @pyqtSignature('')  # prevents actions being handled twice
+    def on_directory_button_tool_clicked(self):
+        """Autoconnect slot activated when directory button is clicked."""
+        # noinspection PyCallByClass,PyTypeChecker
+        # set up parameter from dialog
+        input_path = self.layer.currentLayer().source()
+        input_path = os.path.splitext(input_path)
+        output_name = input_path[0].split('/')[-1]
```

Comment:
`/` is the separator on mac/linux, but not on windows. I guess you will have some issues.

Check in `os.path` package what you want or in QFileInfo, QDir, QFile